python/pyspark/sql/tests/pandas/test_pandas_udf_grouped_agg.py

```diff
@@ -897,6 +897,161 @@ def sum_udf(v):
                     )
                     assert_frame_equal(expected, result)
 
+    def test_iterator_grouped_agg_basic(self):
+        """
+        Test basic functionality of iterator grouped agg pandas UDF with Iterator[pd.Series].
+        """
+        from typing import Iterator
+
+        df = self.spark.createDataFrame(
+            [(1, 1.0), (1, 2.0), (2, 3.0), (2, 5.0), (2, 10.0)], ("id", "v")
+        )
+
+        @pandas_udf("double")
+        def pandas_mean_iter(it: Iterator[pd.Series]) -> float:
+            sum_val = 0.0
+            cnt = 0
+            for series in it:
+                assert isinstance(series, pd.Series)
+                sum_val += series.sum()
+                cnt += len(series)
+            return sum_val / cnt if cnt > 0 else 0.0
+
+        result = df.groupby("id").agg(pandas_mean_iter(df["v"]).alias("mean")).sort("id").collect()
+
+        # Expected means:
+        # Group 1: (1.0 + 2.0) / 2 = 1.5
+        # Group 2: (3.0 + 5.0 + 10.0) / 3 = 6.0
+        expected = [(1, 1.5), (2, 6.0)]
+
+        self.assertEqual(len(result), len(expected))
+        for r, (exp_id, exp_mean) in zip(result, expected):
+            self.assertEqual(r["id"], exp_id)
+            self.assertAlmostEqual(r["mean"], exp_mean, places=5)
```

Comment:
you can directly compare the rows

```suggestion
        expected = [Row... ]
        self.assertEqual(result, expected)
```